src/bp/ui-studio/src/web/views/OneFlow/sidePanel/index.tsx

```diff
@@ -115,9 +110,35 @@ const SidePanelContent: FC<Props> = props => {
   }
 
   const createWorkflow = (topicName: string) => {
-    setSelectedTopic(topicName)
-    setSelectedWorkflow('')
-    setEditWorkflowModalOpen(true)
+    const originalName = 'Workflow'
+    let name = originalName
+    let fullName = buildFlowName({ topic: topicName, workflow: name }, true)
+    let index = 0
+    while (props.flows.find(f => f.name === fullName)) {
+      index++
+      name = `${originalName} ${index}`
+      fullName = buildFlowName({ topic: topicName, workflow: name }, true)
+    }
+
+    setFocusedText(fullName)
+    setNewPath(fullName)
+    props.createFlow(fullName)
+  }
+
+  const createTopic = async () => {
+    const originalName = 'Topic'
+    let name = originalName
+    let index = 0
+    while (props.topics.find(t => t.name === name)) {
```

Comment:
I implemented your solution but then I realized this adds a vulnerability. If you have
- Topic 1
- Topic 2
- Topic 3

And delete Topic 1 it will wrongly create a topic named Topic 2 which already exists.

Reply:
Well then this should work > < 
```
const lastTemplateTopic = props.topic.filter(t => /Topic\s\d/.test(t.name)).pop()
const idx = lastTemplateTopic ? lastTemplateTopic.split(" ")[1] : 0
const name = `Topic ${idx +1}`
```